src/main/scala/com/fulcrumgenomics/util/Sequences.scala

```diff
@@ -150,15 +152,74 @@ object Sequences {
   }
 
   /** Reverse complements a string of bases. */
-  def revcomp(s: String): String = SequenceUtil.reverseComplement(s)
+  def revcomp(s: String): String = {
+    val bs = s.getBytes
+    revcomp(bs)
+    new String(bs)
+  }
 
   /** Returns the sequence that is the complement of the provided sequence. */
   def complement(s: String): String = {
     val bs = s.getBytes
-    forloop(from=0, until=bs.length) { i =>  bs(i) = SequenceUtil.complement(bs(i)) }
+    complement(bs)
     new String(bs)
   }
 
+  /** Reverse complements an array of bases in place. See [[complement()]] for how bases are complemented. */
+  def revcomp(bs: Array[Byte]): Unit = {
+    SequenceUtil.reverse(bs, 0, bs.length)
+    complement(bs)
```

Comment:
I'm torn, since you can't use `SequenceUtil.reverseComplement` since you want IUPAC support, but you make two passes through the array.

Reply:
Eh, fine, I've implemented it as single pass.